sbin/submitty_autograding_shipper.py

```diff
@@ -488,51 +517,110 @@ def unpack_job(which_machine, which_untrusted, next_directory, next_to_grade, ra
             # Close file descriptors
             for var in [fd1, fd2]:
                 if var:
-                    try:
+                    with contextlib.suppress(OSError):
                         os.close(var)
-                    except Exception:
-                        pass
 
-        if not success:
-            return False
+        # If we are waiting, return so now. Otherwise, we can check if
+        # the job returned successfully
+        if status in [GradingStatus.WAITING, GradingStatus.FAILURE]:
+            return status
 
     try:
         with open(local_done_queue_file, 'r') as infile:
             local_done_queue_obj = json.load(infile)
+
         # Check to make certain that the job we received was the correct job
         if random_identifier != local_done_queue_obj['identifier']:
-            success = False
+            status = GradingStatus.WAITING
```

Comment:
I think this line can be wholly removed, since it looks like `GradingStatus.WAITING` is being explicitly returned on line 551.

Reply:
Good catch. Removed.